test/integ_test/evaluator/azureml_eval/utils.py

```diff
@@ -10,89 +10,99 @@
 from torchvision.transforms import ToTensor
 
 from olive.azureml.azureml_client import AzureMLClientConfig
+from olive.common.config_utils import validate_config
+from olive.data.config import DataComponentConfig, DataConfig
 from olive.evaluator.metric import AccuracySubType, LatencySubType, Metric, MetricType
 from olive.systems.azureml import AzureMLDockerConfig, AzureMLSystem
 
 # pylint: disable=redefined-outer-name
 
+_current_dir, _models_dir, _data_dir, _user_script = None, None, None, None
+
 
 def get_directories():
-    current_dir = Path(__file__).resolve().parent
+    global _current_dir, _models_dir, _data_dir, _user_script
```

Comment:
Since pylint is complaining about the global variables, I think you could implement this get_directories as a pytest fixture like in https://github.com/microsoft/Olive/blob/fc4fc903f09da82d3205a79a011a7bf89089f955/test/unit_test/workflows/test_whisper_workflow.py#L23 

If you scope it as "module", it would only run through the contents once if you don't want to be executed multiple times.

Reply:
looks like it's already updated to not use globals

Reply:
Implemented it as recommended by PyLint docs.